examples/src/main/java/com/mapbox/navigation/examples/core/MapboxNavigationActivity.kt

```diff
@@ -494,6 +545,19 @@ class MapboxNavigationActivity : AppCompatActivity() {
         )
     }
 
+    private fun setRoutesPreview(routes: List<NavigationRoute>) {
+        binding.navigateButton.apply {
+            visibility = View.VISIBLE
+            setOnClickListener {
+                visibility = View.GONE
+                setRouteAndStartNavigation(mapboxNavigation.getRoutesPreview()!!.routesList)
```

Comment:
that's a good question! I can imagine that it could be important for UI to receive route line update in such an order that it doesn't need to redraw existing route line. 
For now it's the SDK's client's responsibility to care about it as it up to the clients how to switch between preview and AG. I will add to PR description which specific cases the route preview covers so that we could think about taking more responsibilities from clients to the SDK without loosing flexibility.